hibike/lib/hibike/hibike_device.cpp

```diff
@@ -31,46 +28,56 @@ void hibike_loop() {
     if (read_message(&hibikeBuff) == -1) {
       toggleLED();
     } else {
+      int offset;
       switch (hibikeBuff.messageID) {
 
         case SUBSCRIPTION_REQUEST:
           // change subDelay and send SUB_RESP
-          subDelay = *((uint16_t*) &hibikeBuff.payload[0]);
-          send_subscription_response(&UID, subDelay);
+          subDelay = *((uint16_t*) &hibikeBuff.payload[2]);
+          params = *((uint16_t*) &hibikeBuff.payload[0]);
+          send_subscription_response(params, subDelay, &UID);
           break;
 
         case SUBSCRIPTION_RESPONSE:
           // Unsupported packet
           toggleLED();
           break;
 
-        case DATA_UPDATE:
-          // Unsupported packet
-          toggleLED();
+        case DEVICE_WRITE:
+          //loop over params
+          old_params = params;
+          offset = 2;
+          params = *((uint16_t*)&hibikeBuff.payload[0]);
+          for (uint16_t count = 0; (params >> count) > 0; count++) {
+            if (params & (1<<count)){
+              int status = device_write(count, &hibikeBuff.payload[offset], hibikeBuff.payload_length-offset);
+              if(status){
+                offset += status;}
+              else{
+                params = params & ~(1<<count);}
+              }
+          }
+
+          // params = hibikeBuff.payload[0] - 1;
```

Comment:
FYI I looked over your latest changes, and this comment is currently the only non-ideal thing in the code that I've found.